torch/_higher_order_ops/while_loop.py

```diff
@@ -97,31 +106,31 @@ def body_fn(iter, x):
 
     """
     if torch.compiler.is_dynamo_compiling():
-        return while_loop_op(cond_fn, body_fn, operands)
+        return while_loop_op(cond_fn, body_fn, carried_inputs, tuple())
 
-    def _validate_input(cond_fn, body_fn, operands):
+    def _validate_input(cond_fn, body_fn, carried_inputs):
         if not callable(cond_fn) or not callable(body_fn):
             raise RuntimeError("Expect cond_fn and body_fn to be callbale.")
 
-        if not isinstance(operands, (tuple, list)) or pytree.tree_any(
-            lambda t: not isinstance(t, torch.Tensor), operands
+        if not isinstance(carried_inputs, (tuple, list)) or pytree.tree_any(
+            lambda t: not isinstance(t, torch.Tensor), carried_inputs
         ):
             raise RuntimeError(
-                "Expect operands to be a tuple of possibly nested dict/list/tuple that only"
-                f"consists of tensor leaves, but got {operands}."
+                "Expect carried_inputs to be a tuple of possibly nested dict/list/tuple that only"
+                f"consists of tensor leaves, but got {carried_inputs}."
             )
 
-    _validate_input(cond_fn, body_fn, operands)
+    _validate_input(cond_fn, body_fn, carried_inputs)
 
     with _set_compilation_env(), torch._dynamo.utils.disable_cache_limit():
         return torch.compile(while_loop_op, backend="eager", fullgraph=True)(
-            cond_fn, body_fn, operands
+            cond_fn, body_fn, carried_inputs, tuple()
         )
 
 
 @while_loop_op.py_impl(DispatchKey.CompositeExplicitAutograd)
-def while_loop_dense(cond_fn, body_fn, operands):
-    init_val = operands
+def while_loop_dense(cond_fn, body_fn, carried_inputs, additional_inputs):
+    init_val = carried_inputs
```

Comment:
Nit: by the end of this function, `init_val` are likely no longer initial values. Maybe rename to smth like `carried_vals`?